cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -191,65 +199,101 @@ static String removeReplicasFromConfig(String throttleConfig, Set<String> replic
     return String.join(",", throttles);
   }
 
-  private void removeLeaderThrottledReplicasFromTopic(Properties config, String topic, Set<String> replicas) {
-    String oldLeaderThrottledReplicas = config.getProperty(LEADER_THROTTLED_REPLICAS);
-    if (oldLeaderThrottledReplicas != null) {
+  /**
+   * It gets whether there is any throttled leader replica specified in the configuration property. If there is and the
+   * specified throttled leader replica does not equal to "*", it modifies the configuration property by removing a
+   * given set of replicas from the a set of throttled leader replica
+   *
+   * @param config configuration properties
+   * @param topic name of topic which contains <code>replicas</code>
+   * @param replicas replicas to remove from the configuration properties
+   * @return true if the given configuration properties are modified and false otherwise
+   */
+  private boolean removeLeaderThrottledReplicasFromTopic(Properties config, String topic, Set<String> replicas) {
+    String currLeaderThrottledReplicas = config.getProperty(LEADER_THROTTLED_REPLICAS);
+    if (currLeaderThrottledReplicas != null) {
+      if (currLeaderThrottledReplicas.equals(WILD_CARD_ASTERISK)) {
+        LOG.debug("Existing config throttles all leader replicas. So, not remove any leader replica throttle");
```

Comment:
Nit: `So, not remove ...` -> `So, do not remove ...`
- multiple occurrences, please see `#removeThrottledRateFromBroker `